indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -353,16 +348,86 @@ private TaskLockPosse createOrFindLockPosse(
     giant.lock();
 
     try {
-      final String dataSource = task.getDataSource();
-      final int priority = task.getPriority();
+      return createOrFindLockPosse(
+          lockType,
+          task.getId(),
+          task.getGroupId(),
+          task.getDataSource(),
+          interval,
+          preferredVersion,
+          task.getPriority(),
+          false
+      );
+    }
+    finally {
+      giant.unlock();
+    }
+  }
+
+  @Nullable
+  private TaskLockPosse createOrFindLockPosse(Task task, TaskLock taskLock)
+  {
+    giant.lock();
+
+    Preconditions.checkArgument(
```

Comment:
this will throw and never unlock!

Reply:
OMG, thank you for finding! I’ll fix it soon.